packages/core/content-releases/admin/src/components/EntryValidationPopover.tsx

```diff
@@ -0,0 +1,376 @@
+}) => {
+  if (!contentTypeHasReviewWorkflow) {
+    return formatMessage({
+      id: 'content-releases.pages.ReleaseDetails.entry-validation.review-stage.not-enabled',
+      defaultMessage: 'This entry is not associated to any workflow.',
+    });
+  }
+
+  if (requiredStage && requiredStage.id !== entryStage?.id) {
+    return formatMessage(
+      {
+        id: 'content-releases.pages.ReleaseDetails.entry-validation.review-stage.not-ready',
+        defaultMessage: 'This entry is not at the required stage for publishing. ({stageName})',
+      },
+      {
+        stageName: requiredStage?.name ?? '',
+      }
+    );
+  }
+  return formatMessage(
+    {
+      id: 'content-releases.pages.ReleaseDetails.entry-validation.review-stage.ready',
+      defaultMessage: 'This entry is at the required stage for publishing. ({stageName})',
+    },
+    {
+      stageName: requiredStage?.name ?? '',
+    }
+  );
+};
+
+const ReviewStageValidation = ({
+  contentTypeHasReviewWorkflow,
+  requiredStage,
+  entryStage,
+}: {
+  contentTypeHasReviewWorkflow: boolean;
+  requiredStage?: Stage;
+  entryStage?: Stage;
+}) => {
+  const { formatMessage } = useIntl();
+
+  const Icon = getReviewStageIcon({
+    contentTypeHasReviewWorkflow,
+    requiredStage,
+    entryStage,
+  });
+
+  return (
+    <Flex direction="column" gap={1} width="100%" padding={5}>
+      <Flex gap={2} width="100%">
+        <Typography fontWeight="bold">
+          {formatMessage({
+            id: 'content-releases.pages.ReleaseDetails.entry-validation.review-stage',
+            defaultMessage: 'Review stage',
+          })}
+        </Typography>
+        {Icon}
+      </Flex>
+      <Typography>
+        {getReviewStageMessage({
+          contentTypeHasReviewWorkflow,
+          requiredStage,
+          entryStage,
+          formatMessage,
+        })}
+      </Typography>
+    </Flex>
+  );
+};
+
+export const EntryValidationPopover = ({
+  action,
+  schema,
+  entry,
+  status,
+}: EntryValidationPopoverProps) => {
+  const { validate, isLoading } = unstable_useDocument(
+    {
+      collectionType: schema?.kind ?? '',
+      model: schema?.uid ?? '',
+    },
+    {
+      // useDocument makes a request to get more data about the entry, but we only want to have the validation function so we skip the request
+      skip: true,
+    }
+  );
+
+  // Validation errors
+  const errors = isLoading ? null : validate(entry);
+  const hasErrors = errors ? Object.keys(errors).length > 0 : false;
+
+  // Entry stage
+  const contentTypeHasReviewWorkflow = schema?.options?.reviewWorkflows ?? false;
+  const requiredStage = schema?.requiredStageToPublish;
+  const entryStage = entry.strapi_stage;
+
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <Popover.Root>
+      <EntryStatusTrigger
+        action={action}
+        status={status}
+        hasErrors={hasErrors}
+        requiredStage={requiredStage}
+        entryStage={entryStage}
+      />
```

Comment:
Why do you forward the ref in the EntryStatusTrigger definition if you don't use it?